test/e2e_node/device_plugin_test.go

```diff
@@ -236,9 +236,133 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
+			gomega.Expect(devID1).To(gomega.Not(gomega.Equal("")))
+
+			pod1, err := e2epod.NewPodClient(f).Get(ctx, pod1.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Restarting Kubelet")
+			restartKubelet(true)
+
+			ginkgo.By("Wait for node to be ready again")
+			e2enode.WaitForAllNodesSchedulable(ctx, f.ClientSet, 5*time.Minute)
+
+			ginkgo.By("Waiting for the pod to fail with admission error as device plugin hasn't re-registered yet")
+			gomega.Eventually(ctx, getPod).
+				WithArguments(f, pod1.Name).
+				WithTimeout(time.Minute).
+				Should(HaveFailedWithAdmissionError(),
+					"the pod succeeded to start, when it should fail with the admission error")
+
+			ginkgo.By("Confirming that after kubelet restart, fake-device assignment is kept")
+			devIDRestart1 := parseLog(ctx, f, pod1.Name, pod1.Name, deviceIDRE)
+			framework.ExpectEqual(devIDRestart1, devID1)
+		})
+
+		ginkgo.It("Keeps device plugin assignments across pod and kubelet restarts (no device plugin re-registration)", func(ctx context.Context) {
+			podRECMD := "devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs && sleep 60"
+			pod1 := e2epod.NewPodClient(f).CreateSync(ctx, makeBusyboxPod(SampleDeviceResourceName, podRECMD))
+			deviceIDRE := "stub devices: (Dev-[0-9]+)"
+			devID1 := parseLog(ctx, f, pod1.Name, pod1.Name, deviceIDRE)
+			gomega.Expect(devID1).To(gomega.Not(gomega.Equal("")))
+
+			pod1, err := e2epod.NewPodClient(f).Get(ctx, pod1.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Wait for node to be ready again")
+			e2enode.WaitForAllNodesSchedulable(ctx, f.ClientSet, 5*time.Minute)
+
+			ginkgo.By("Waiting for container to restart")
+			ensurePodContainerRestart(ctx, f, pod1.Name, pod1.Name)
+
+			ginkgo.By("Confirming that after a container restart, fake-device assignment is kept")
+			devIDRestart1 := parseLog(ctx, f, pod1.Name, pod1.Name, deviceIDRE)
+			framework.ExpectEqual(devIDRestart1, devID1)
+
+			ginkgo.By("Restarting Kubelet")
+			restartKubelet(true)
+
+			ginkgo.By("Wait for node to be ready again")
+			e2enode.WaitForAllNodesSchedulable(ctx, f.ClientSet, 5*time.Minute)
+
+			ginkgo.By("Waiting for the pod to fail with admission error as device plugin hasn't re-registered yet")
+			gomega.Eventually(ctx, getPod).
+				WithArguments(f, pod1.Name).
+				WithTimeout(time.Minute).
+				Should(HaveFailedWithAdmissionError(),
+					"the pod succeeded to start, when it should fail with the admission error")
+
+			ginkgo.By("Confirming that after a kubelet restart, fake-device assignment is kept")
+			devIDRestart1 = parseLog(ctx, f, pod1.Name, pod1.Name, deviceIDRE)
+			framework.ExpectEqual(devIDRestart1, devID1)
+		})
+
+		ginkgo.It("Keeps device plugin assignments after the device plugin has been re-registered (no kubelet or pod restart)", func(ctx context.Context) {
+			podRECMD := "devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs && sleep 60"
```

Comment:
do we want to increase the sleep period here, effectively making "forever" for the test timescale?

Reply:
Sure, we should definitely do that.